### PR TITLE
refactor(github): improve github collection performance

### LINKED ISSUE #1230: Improve github performance - pull_request_commits
## Description
Now, it's very slow to collect pull_request_commits, even it has only several records, but the process makes dozens of requests for empty pages

## Describe the solution you'd like
After discussion with @klesh , we can set the the number of concurrent to 1 when collecting this kind of data

## Has the Feature been Requested Before?
No

## Additional context
We also need to check if the process of collecting comments, events can be improved

### PR DESCRIPTION
**Pull request by warren830**
# Summary

Improve github collection performance

### Key Points

- [ ] This is a breaking change
- [ ] New or existing documentation is updated

### Description
We can get page information from github api headers

### Does this close any open issues?
Closes #1230 #1220 #1221

### Current Behavior
Keep trying to get pages, sometimes even request for over 50 pages but only first page has data
Iterated issues one by one to get events/comments

### New Behavior
Only request for the pages with data
Get events/comments by repo

### Screenshots
Include any relevant screenshots here.

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

File: plugins/github/tasks/github_pull_request_commits_collector.go [hunk 1]
```diff
@@ -36,22 +36,27 @@ type PullRequestCommit struct {
...
-	githubPr := &models.GithubPullRequest{}
 	for cursor.Next() {
+		githubPr := &models.GithubPullRequest{}
 		err = lakeModels.Db.ScanRows(cursor, githubPr)
 		if err != nil {
 			return err
 		}
-		err = ProcessCollection(owner, repo, githubPr, scheduler, apiClient)
+		err = scheduler.Submit(func() error {
+			processErr := ProcessCollection(owner, repo, githubPr, scheduler, apiClient)
```
Comment (klesh) [inline plugins/github/tasks/github_pull_request_commits_collector.go:49]: nested queue might cause deadlock when outer jobs are waiting for their inner jobs at the end of the queue to finish.

File: plugins/github/tasks/github_api_client.go [hunk 2]
```diff
@@ -45,66 +46,55 @@ func CreateApiClient(endpoint string, tokens []string, ctx context.Context) *Git
...
+			return nil
+		})
+		if err != nil {
+			return err
 		}
-		step += 1
 	}
 
-	scheduler.WaitUntilFinish()
+	githubApiClient.Scheduler.WaitUntilFinish()
```
[the remark this answers is not in the document]
Comment (warren830) [reply plugins/github/tasks/github_api_client.go:98]: That would be more fancy! I will fix it in the next push

File: errors/errors.go [hunk 3]
```diff
@@ -23,3 +26,4 @@ func NewError(status int, message string) *Error {
 }
 
 var InternalError = NewError(http.StatusInternalServerError, "Server Internal Error")
+var TaskCanceled = fmt.Errorf("task got canceled")
```
[the remark this answers is not in the document]
Comment (warren830) [reply errors/errors.go:29]: In utils.worker_scheduler, there is logger.Error("task got canceled", errors.TaskCanceled), and now we import this in plugins/core/apiclient.go, this caused cycle import

**Review by klesh — APPROVED** [review]
LGTM

**Pull request merged**